### ISSUE TITLE
renamed aggregate of existing volumes causes trident to fail

### ISSUE THREAD

**Issue by lopf** [NetApp/trident#229]
We recently migrated volumes to another aggregate and then noticed that trident was unable to provision new PVs. Trident was also unable to initialize once we tried to reinstall.
```
time="2019-03-26T10:52:35Z" level=warning msg="Kubernetes frontend couldn't provision a volume: Trident initialization failed; the assigned aggregates for SVM mysvm do not include the configured aggregate aggr_old (will retry upon resync)" volume=myvol-develop-mysql-pvc-15cd6
```
While it is obvious that it's because of the static backend aggregate definition, we found no way to update it. Data is stored in etcd. Is there a way to update etcd to search&replace the aggregate name? Or is there another solution or strategy for our problem?

**Comment by kangarlou** [comment]
Hi @lopf. Have you tried updating the backend definition via `tridentctl update backend`? Once you update the backend, Trident should use the new aggregate for provisioning volumes and the old volumes would also be handled correctly (e.g., they can be resized, deleted).

**Comment by lopf** [comment]
Thank you, we actually tried to `update backend` but it failed. Now we checked again and found a slightly incorrect documentation.

It tells that `backendName` consists of "Driver name + “_” + dataLIF" by default. So we naturally tried `ontap-nas_<ip>` but the correct name would have been `ontapnas_<ip>`, without a dash in the driver name.

https://netapp-trident.readthedocs.io/en/stable-v18.10/kubernetes/operations/tasks/backends/ontap.html#backend-configuration-options

We'll update the ticket once we gave it another try.
